lib/shoulda/matchers/active_record/association_matcher.rb

```diff
@@ -1267,6 +1331,12 @@ def foreign_key_exists?
           !(belongs_foreign_key_missing? || has_foreign_key_missing?)
         end
 
+        def foreign_type_matches?
+          return true unless options.key?(:foreign_type)
```

Comment:
We don't typically return early or use the modifier form of `unless`. Since the end result is a boolean, what do you think about combining these into one conditional?

``` ruby
!options.key?(:foreign_type) || (
  !belongs_foreign_type_missing? &&
  !has_foreign_type_missing?
)
```